src/platform/Linux/PlatformManagerImpl.cpp

```diff
@@ -44,15 +45,68 @@ namespace DeviceLayer {
 
 PlatformManagerImpl PlatformManagerImpl::sInstance;
 
+namespace {
+
+void SignalHandler(int signum)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    ChipLogDetail(DeviceLayer, "Caught signal %d", signum);
+
+    // The BootReason attribute SHALL indicate the reason for the Node’s most recent boot, the real usecase
+    // for this attribute is embedded system. In Linux simulation, we use different signals to tell the current
+    // running process to terminate with different reasons.
+    switch (signum)
+    {
+    case SIGINT:
+        ConfigurationMgrImpl().StoreBootReasons(EMBER_ZCL_BOOT_REASON_TYPE_SOFTWARE_RESET);
+        err = CHIP_ERROR_REBOOT_SIGNAL_RECEIVED;
+        break;
+    case SIGHUP:
+        ConfigurationMgrImpl().StoreBootReasons(EMBER_ZCL_BOOT_REASON_TYPE_BROWN_OUT_RESET);
+        err = CHIP_ERROR_REBOOT_SIGNAL_RECEIVED;
+        break;
+    case SIGTERM:
+        ConfigurationMgrImpl().StoreBootReasons(EMBER_ZCL_BOOT_REASON_TYPE_POWER_ON_REBOOT);
+        err = CHIP_ERROR_REBOOT_SIGNAL_RECEIVED;
+        break;
+    case SIGQUIT:
+        ConfigurationMgrImpl().StoreBootReasons(EMBER_ZCL_BOOT_REASON_TYPE_HARDWARE_WATCHDOG_RESET);
+        err = CHIP_ERROR_REBOOT_SIGNAL_RECEIVED;
+        break;
+    case SIGABRT:
+        ConfigurationMgrImpl().StoreBootReasons(EMBER_ZCL_BOOT_REASON_TYPE_SOFTWARE_WATCHDOG_RESET);
+        err = CHIP_ERROR_REBOOT_SIGNAL_RECEIVED;
+        break;
+    case SIGBUS:
```

Comment:
Will this prevent core file capture and whatnot on SIGBUS?

Reply:
I adjusted to use the following non-critical signals for reboot reason simulation 
    sigaction(SIGINT, &action, NULL);
    sigaction(SIGHUP, &action, NULL);
    sigaction(SIGTERM, &action, NULL);
    sigaction(SIGUSR1, &action, NULL);
    sigaction(SIGUSR2, &action, NULL);
    sigaction(SIGTSTP, &action, NULL);